client/app/telecom/pack/slots/hubic/pack-hubic.controller.js

```diff
@@ -25,67 +28,72 @@ angular.module("managerApp").controller("PackHubicCtrl", function (OvhApiPackXds
                 errorRule: {
                     status: "error"
                 },
-                scope: $scope.id
+                scope: this.$scope.id
             });
     }
 
     /**
      * Load all hubic services
      * @return {Promise}
      */
-    this.loadHubics = function () {
+    loadHubics () {
         this.loaders.services = true;
         this.services = [];
 
-        return OvhApiPackXdslHubic.Aapi().query({
-            packId: $stateParams.packName
-        }).$promise.then(function (services) {
-            self.services = _.map(services, function (service) {
-                var voucherUrl = [URLS.hubicVoucher, "token=" + service.voucher].join("?");
+        return this.OvhApiPackXdslHubic.Aapi().query({
+            packId: this.$stateParams.packName
+        }).$promise.then((services) => {
+            this.services = _.map(services, (service) => {
+                let voucherUrl = [this.URLS.hubicVoucher, "token=" + service.voucher].join("?");
                 return _.extend(
                     service,
                     {
-                        url: service.voucher ? voucherUrl : URLS.hubicLogin
+                        url: service.voucher ? voucherUrl : this.URLS.hubicLogin
                     }
                 );
             });
 
-            var servicesCodeUsed = _.filter(self.services, { isUsed: true });
+            let servicesCodeUsed = _.filter(this.services, { isUsed: true });
```

Comment:
you could use `const`.

Reply:
each item of this array which promise succeed, will have a modification of email prop, so does it need to be a const ?